### PR TITLE
K8SPSMDB-408, K8SPSMDB-401: Fix arbiter liveness probe

### PR DESCRIPTION
**Pull request by blez**
[K8SPSMDB-408](https://jira.percona.com/browse/K8SPSMDB-408) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

File: pkg/controller/perconaservermongodb/mgo.go [hunk 1]
```diff
@@ -199,14 +198,14 @@ func (r *ReconcilePerconaServerMongoDB) reconcileCluster(cr *api.PerconaServerMo
 	return clusterInit, nil
 }
 
-func inShard(client *mgo.Client, podName string) (bool, error) {
+func inShard(client *mgo.Client, rsName string) (bool, error) {
 	shardList, err := mongo.ListShard(context.TODO(), client)
```
Comment (delgod) [inline pkg/controller/perconaservermongodb/mgo.go:202]: can we use `getShardMap` instead?
I want to be 100% sure that all Pods are added to Sharding

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/controller/perconaservermongodb/mgo.go (hunk 1 above)
Comment (blez) [reply pkg/controller/perconaservermongodb/mgo.go:202]: We do it here 
https://github.com/percona/percona-server-mongodb-operator/blob/main/pkg/controller/perconaservermongodb/mgo.go#L127-L176

In any case we did not check all the pods before, only the first one.
https://github.com/percona/percona-server-mongodb-operator/blob/main/pkg/controller/perconaservermongodb/mgo.go#L101

**Review by delgod — APPROVED** [review]

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| self-healing | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| operator-self-healing | passed |
| rs-shard-migration | passed |
| arbiter | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| upgrade | failed |
| version-service | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/590/commits/b6edff69a3b9f2745e0d4dec019598515f6ea365
image: `perconalab/percona-server-mongodb-operator:PR-590-b6edff69`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| limits | passed |
| arbiter | passed |
| scaling | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| version-service | passed |
| upgrade-sharded | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/590/commits/e92bbe898a9de3edb7b37d6b36a894fc32703bc4
image: `perconalab/percona-server-mongodb-operator:PR-590-e92bbe89`

**Review by delgod — APPROVED** [review]

**Pull request merged**